### PR TITLE
Ec dbm cryto merge with dev

### PR DESCRIPTION
**Pull request by dbmcclain**
Huge changes to crypto substrate, addition of Randhound sim, but very few changes to running sources from dev branch. Some simple additions to cost-construction and cosi-handlers to deal with Randhound message delivery and state requirements.

Clean compiles, and runs (emotiq:main) okay... (I think?)

### CODE REVIEW

**Comment by dbmcclain** [comment]
hmm... fixed the broken crypto tests. Runs just fine here on LWM and CCL/Mac. But CI gives an error on CCL/Linux.... Trying to track down if it is real, or just a CI quirk

File: src/Cosi-BLS/cosi-netw-xlat.lisp [hunk 1]
```diff
@@ -91,6 +91,15 @@ THE SOFTWARE.
       (apply 'ac:send (node-self *my-node*) msg)
     (gossip-send pkey nil msg)))
         
+;; --------------------------------------------------------------
```
Comment (easye) [inline src/Cosi-BLS/cosi-netw-xlat.lisp]: @dbmcclain There should be no references to IP addresses or ports.

**Comment by dbmcclain** [comment]
We have a clean merge with dev branch now. AFAIK there are no issues with making this merge for new crypto. Randhound now has issues with Gossip, now that `*use-real-gossip*` is gone. But that is a separate task for me now.

**Comment by dbmcclain** [comment]
somehow old code snuck back in there in cosi-netw-xlat. Now removed and try again.

**Comment by dbmcclain** [comment]
Getting consistent CI failures (across 2 tries) with generating genesis blocks in :TEST-GENERATE, but works just fine on my local Linux system from clean start. ???

**Comment by dbmcclain** [comment]
Finally got a CI pass

**Review by guitarvydas — APPROVED** [review]
passes CI
passes simple local test (emotiq:main)
merges with today's (Aug. 5) dev

**Pull request merged**